platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/SyncGossip.java

```diff
@@ -190,9 +191,16 @@ public SyncGossip(
                 connectionManagers::newConnection,
                 platformContext.getTime());
         // allow other members to create connections to me
-        final Address address = addressBook.getAddress(selfId);
+        final RosterEntry rosterEntry = RosterUtils.getRosterEntry(roster, selfId.id());
+        // Assume all ServiceEndpoints use the same port and use the port from the first endpoint.
+        // Previously, this code used a "local port" corresponding to the internal endpoint,
+        // which should normally be the second entry in the endpoints list if it's obtained via
+        // a regular AddressBook -> Roster conversion.
+        // The assumption must be correct, otherwise, if ports were indeed different, then the old code
+        // using the AddressBook would never have listened on a port associated with the external endpoint,
+        // thus not allowing anyone to connect to the node from outside the local network, which we'd have noticed.
```

Comment:
This isn't a completely new behavior introduced by the refactoring though. The legacy code would just listen on the `Address.getListenPort()`, and the latter would return the result of `getPortInternal()`. In other words, the current code assumes that the port number is the same for both internal and external endpoints.

Since Roster doesn't differentiate between "internal" and "external" endpoints, this same assumptions simply transforms here to assuming that any endpoint defined in the RosterEntry uses the same port, and hence using the one from the entry with index 0 is equivalent to the old code using the internal port number.

So this refactoring doesn't really make this part any different from the current code behavior.

I do agree that this may not be the best strategy, especially if different endpoints may want to use different port numbers in the future, which would require a node to open several server sockets on different ports, and preferably bind them to specific IP addresses of the node per the service endpoint definitions. However, overhauling this part is completely outside the scope of this PR or the Roster or TSS projects as a whole. Please open a separate ticket if you feel strongly about the overhaul.